modules/tinymce/src/themes/silver/test/ts/browser/view/ViewTest.ts

```diff
@@ -0,0 +1,337 @@
+import { ApproxStructure, Assertions, Mouse, StructAssert, TestStore, UiFinder, Waiter } from '@ephox/agar';
+import { context, describe, it } from '@ephox/bedrock-client';
+import { Arr, Fun } from '@ephox/katamari';
+import { Attribute, Css, Html, SugarBody } from '@ephox/sugar';
+import { TinyApis, TinyDom, TinyHooks, TinySelections, TinyUiActions } from '@ephox/wrap-mcagar';
+import { assert } from 'chai';
+
+import Editor from 'tinymce/core/api/Editor';
+import { View } from 'tinymce/core/api/ui/Ui';
+
+describe('browser.tinymce.themes.silver.view.ViewTest', () => {
+  context('Iframe mode', () => {
+    const store = TestStore();
+    const hook = TinyHooks.bddSetup<Editor>({
+      base_url: '/project/tinymce/js/tinymce',
+      toolbar_mode: 'floating',
+      toolbar: Arr.range(10, Fun.constant('bold | italic ')).join(''),
+      width: 500,
+      setup: (editor: Editor) => {
+        const injectAndLog = (name: string, html: string = '') => (api: View.ViewInstanceApi) => {
+          api.getContainer().innerHTML = html;
+          store.add(name);
+        };
+
+        editor.ui.registry.addView('myview1', {
+          buttons: [
+            {
+              type: 'button',
+              text: 'Button 1',
+              onAction: store.adder('myview1:button1')
+            },
+            {
+              type: 'button',
+              text: 'Button 2',
+              onAction: store.adder('myview1:button2'),
+              buttonType: 'primary'
+            }
+          ],
+          onShow: (api) => {
+            api.getContainer().innerHTML = '<button>myview1</button>';
+            api.getContainer().querySelector('button')?.focus();
+            store.add('myview1:show');
+          },
+          onHide: injectAndLog('myview1:hide')
+        });
+
+        editor.ui.registry.addView('myview2', {
+          buttons: [
+            {
+              type: 'button',
+              text: 'Button 1',
+              onAction: store.adder('myview2:button1'),
+              buttonType: 'secondary'
+            },
+            {
+              type: 'button',
+              text: 'Button 2',
+              onAction: store.adder('myview2:button2'),
+              buttonType: 'primary'
+            }
+          ],
+          onShow: injectAndLog('myview2:show', 'myview2'),
+          onHide: injectAndLog('myview2:hide')
+        });
+
+        editor.ui.registry.addView('myview3', {
+          onShow: injectAndLog('myview3:show', 'myview3'),
+          onHide: injectAndLog('myview3:hide')
+        });
+
+        editor.ui.registry.addContextToolbar('test-context', {
+          predicate: (node) => node.nodeName.toLowerCase() === 'img',
+          items: 'bold'
+        });
+      }
+    }, []);
+
+    const clickViewButton = (editor: Editor, tooltip: string) => {
+      const btn = UiFinder.findIn(TinyDom.container(editor), `.tox-view button[title='${tooltip}']`).getOrDie();
```

Comment:
Could we use `TinyUiActions.clickOnUi` here?

Reply:
Done